lsp-mode.el

```diff
@@ -262,7 +264,12 @@ Optional arguments:
 `:prefix-function' is a function called for getting the prefix for completion.
  The function takes no parameter and returns a cons (start . end) representing
  the start and end bounds of the prefix. If it's not set, the client uses a
- default prefix function."
+ default prefix function.
+
+`:port-fn' is a function called after starting the language server process, but
+ before connecting to it. The function must return a number, which will be used
+ as the port to connect to. This is useful for servers which use random port
+ numbers, and also for delaying until the server is ready to accept connections."
```

Comment:
Isn't this too hacky? Instead, we can replace this with a way to defer requests, or wait for when the server starts accepting connections/input.

Reply:
Agreed using this to delay the connection is super hacky and not the ideal goal. I can change the docs to not mention that and work on a better asynchronous solution. We do need this function anyways to handle dynamic port numbers.